src/core/server/errors/index.ts

```diff
@@ -679,3 +679,11 @@ export class PasswordIncorrect extends CoralError {
     });
   }
 }
+
+export class CannotUpdateLocalProfile extends CoralError {
```

Comment:
I wouldn't bother throwing a new `CoralError` for this, as it's only going to occur if a user tries to call this API directly. Throwing `new Error("cannot update profile, settings disallowed")` or something should be fine.